org.openscenegraph.android/jni/JNIosg.cpp

```diff
@@ -139,6 +178,931 @@ JNIEXPORT jint JNICALL Java_org_openscenegraph_osg_core_Group_nativeGetNumChildr
+		}
+	}
+
+	if(g!=NULL)
+	{
+		g->setTexCoordArray((int)tex_unit,_array);
+	}
+}
+
+JNIEXPORT void JNICALL Java_org_openscenegraph_osg_core_Geometry_nativeSetTexCoordArrayNative(JNIEnv* env, jclass, jlong cptr, jlong array_cptr, jint tex_unit)
+{
+	osg::Geometry *g = reinterpret_cast<osg::Geometry*>(cptr);
+	osg::Vec2Array* a = reinterpret_cast<osg::Vec2Array*>(array_cptr);
+
+	if( (g!=NULL) && (a!=NULL) )
+	{
+		g->setTexCoordArray((int)tex_unit,a);
+	}
+}
+
+JNIEXPORT jlong JNICALL Java_org_openscenegraph_osg_core_Geometry_nativeGetTexCoordArray(JNIEnv* env, jclass, jlong cptr, jint tex_unit)
+{
+	jlong result = 0;
+	osg::Geometry* g = reinterpret_cast<osg::Geometry*>(cptr);
+	if(g!=NULL)
+	{
+		osg::Vec2Array* a = reinterpret_cast<osg::Vec2Array*>(g->getTexCoordArray((int)tex_unit));
+		result = reinterpret_cast<jlong>(a);
+	}
+	return result;
+}
+
+JNIEXPORT void JNICALL Java_org_openscenegraph_osg_core_Geometry_nativeSetNormalBinding(JNIEnv *, jclass, jlong cptr, jbyte binding)
+{
+	osg::Geometry *g = reinterpret_cast<osg::Geometry*>(cptr);
+	g->setNormalBinding(osg::Geometry::AttributeBinding(binding));
+}
+
+JNIEXPORT void JNICALL Java_org_openscenegraph_osg_core_Geometry_nativeSetColorBinding(JNIEnv *, jclass, jlong cptr, jbyte binding)
+{
+	osg::Geometry *g = reinterpret_cast<osg::Geometry*>(cptr);
+	g->setColorBinding(osg::Geometry::AttributeBinding(binding));
+}
+
+JNIEXPORT jboolean JNICALL Java_org_openscenegraph_osg_core_Geometry_nativeAddPrimitiveSet(JNIEnv *, jclass, jlong cptr, jlong set_cptr)
+{
+	osg::Geometry *g = reinterpret_cast<osg::Geometry*>(cptr);
+	osg::PrimitiveSet *n = reinterpret_cast<osg::PrimitiveSet*>(set_cptr);
+	if(g!=NULL)
+	{
+        return g->addPrimitiveSet(n)?JNI_TRUE:JNI_FALSE;
+	}
+	return JNI_FALSE;
+}
+
+JNIEXPORT void JNICALL Java_org_openscenegraph_osg_core_Geometry_nativeSetPrimitiveSetList(JNIEnv* env, jclass, jlong cptr, jlong list_cptr)
+{
+	osg::Geometry* g = reinterpret_cast<osg::Geometry*>(cptr);
+	osg::Geometry::PrimitiveSetList* n = reinterpret_cast<osg::Geometry::PrimitiveSetList*>(list_cptr);
+	if(g!=NULL)
+	{
+		g->setPrimitiveSetList(*n);
+	}
+}
+
+/*
+ * osg::Drawable
+ */
+JNIEXPORT void JNICALL Java_org_openscenegraph_osg_core_Drawable_nativeDispose(JNIEnv* env, jclass, jlong cptr)
+{
+	osg::Drawable* g = reinterpret_cast<osg::Drawable*>(cptr);
+	if(g!=NULL)
+		g->unref();
+}
+
+JNIEXPORT jlong JNICALL Java_org_openscenegraph_osg_core_Drawable_nativeAsDrawable(JNIEnv *, jclass, jlong cptr)
+{
+	osg::Drawable *g = reinterpret_cast<osg::Drawable *>(cptr);
+	return reinterpret_cast<jlong>(g);
+}
+
+/*
+ * osg::PrimitiveSet
+ */
+JNIEXPORT void JNICALL Java_org_openscenegraph_osg_core_PrimitiveSet_nativeDispose(JNIEnv *, jclass, jlong cptr)
+{
+	osg::PrimitiveSet *g = reinterpret_cast<osg::PrimitiveSet *>(cptr);
+	if(g!=NULL)
+		g->unref();
+}
+
+/*
+JNIEXPORT jlong JNICALL Java_org_openscenegraph_osg_core_PrimitiveSet_nativeCreatePrimitiveSet(JNIEnv *, jclass, jint type, jint mode)
```

Comment:
Better if we do not leave dead code, if it does not work/serve anymore is better to remove it.